Tests/SentryTests/SentryBinaryImageCache.swift

```diff
@@ -0,0 +1,121 @@
+import XCTest
+
+class SentryBinaryImageCacheTests: XCTestCase {
+    var sut: SentryBinaryImageCache {
+        SentryBinaryImageCache.shared
+    }
+
+    override func setUp() {
+        super.setUp()
+        sut.start()
+    }
+
+    override func tearDown() {
+        sut.stop()
+        super.tearDown()
+    }
+
+    func testBinaryImageAdded() {
+        var binaryImage0 = createCrashBinaryImage(0)
+        var binaryImage1 = createCrashBinaryImage(100)
+        var binaryImage2 = createCrashBinaryImage(200)
+        var binaryImage3 = createCrashBinaryImage(400)
+        sut.binaryImageAdded(&binaryImage1)
+
+        XCTAssertEqual(sut.cache.count, 1)
+        XCTAssertEqual(sut.cache.first?.name, "Expected Name at 100")
+
+        sut.binaryImageAdded(&binaryImage3)
+        XCTAssertEqual(sut.cache.count, 2)
+        XCTAssertEqual(sut.cache.last?.name, "Expected Name at 400")
+
+        sut.binaryImageAdded(&binaryImage2)
+        XCTAssertEqual(sut.cache.count, 3)
+        XCTAssertEqual(sut.cache.first?.name, "Expected Name at 100")
+        XCTAssertEqual(sut.cache[1].name, "Expected Name at 200")
+        XCTAssertEqual(sut.cache.last?.name, "Expected Name at 400")
```

Comment:
`m`: Instead of doing manual comparisons, we could simply call `soft` on the cache and compare the lists with something like
```
let sorted = sut.cache.sorted {
    return $0.address < $1.address
}

XCTAssertEqual(sorted, sut.cache)
```

Reply:
Well, this test is not meant to change anymore, doesn't  matter the search algorithm, so Im fine with this direct approach.

Reply:
> this test is not meant to change anymore

I don't really understand that part, but we can also keep it as it is.

Reply:
What I mean is that does not make sense for this test to ever change now that it works, so I would avoid preemptively refactoring 😁